### ISSUE TITLE
What causes confabulations?

### ISSUE THREAD

**Issue by bigbrother001** [jlko/semantic_uncertainty#3]
Thanks for great work! The method that cluster generations by bidirectional entailment is inspiring. 

However, it seems that the paper does not have an in-depth analysis of the causes of confabulations. The paper only mentions it because random seed. Could you provide more explanations of what causes confabulations (besides random seeds)?

**Comment by CarhoJohn** [comment]
@bigbrother001 
> Thanks for great work! The method that cluster generations by bidirectional entailment is inspiring.
> 
> However, it seems that the paper does not have an in-depth analysis of the causes of confabulations. The paper only mentions it because random seed. Could you provide more explanations of what causes confabulations (besides random seeds)?

Hi, the paper has mentioned that, 
> when LLMs are consistently wrong as a result of being trained on erroneous data such as common misconceptions; when the LLM ‘lies’ in pursuit of a reward; or systematic failures of reasoning or generalization.

These may be some of the reasons generating "confabulations". Though there are probably other sources, I think this is not the main research question in this paper.

**Comment by bigbrother001** [comment]
@CarhoJohn 
Maybe I have not ask the question clearly.

The paper states: 
>We distinguish this from cases in which a similar 'symptom' is caused by the following different mechanisms: when LLMs are consistently wrong as a result of being trained on erroneous data such as common misconceptions;  when the LLM ‘lies’ in pursuit of a reward;  or systematic failures of reasoning or generalization.

Isn't this to say that the problems caused by these mechanisms are not a kind of confabulations?

I wanted to ask that what causes confabulations, rather than what causes hallucinations (It is a superset of confabulations).

Looking forward to your reply.

**Comment by jlko** [comment]
@CarhoJohn: Thanks for helping out!

@bigbrother001: Thanks for your interest in our paper.

You are right that, as far as I remember, the paper does not discuss 'causes' of confabulations. We observe them empirically and propose a method to detect them.

I can offer my own speculations as to what causes confabulations, but note that these are speculations and not backed up by concrete evidence. I think (and I think this has been discussed in prior work as well) that the pre-training objective encourages confabulatory behavior. The next-token prediction objective encourages the model to predict 'plausible' next tokens. I.e., given the input 'When was the Mona Lisa painted? In the year', it makes sense for the model to learn to assign probability mass across a set of 'plausible' years if it does not already 'know' the answer exactly due to the nature of the negative log likelihood loss function. Sampling from this probability distribution at test time will then lead to confabulations. Conversely, it does not make sense for the model to predict 'I don't know' during pre-training as this will almost never appear in the pre-training data.

In the future, it would be great if you could send questions about the paper and not the codebase to the email address given in the paper!
